### PR TITLE
Save Hadoop conf file instead of appending all of them to connection url

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Save Hadoop conf file instead of appending all of them to connection url

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala [hunk 1]
```diff
@@ -104,6 +109,24 @@ class MiniYarnService(name: String) extends AbstractService(name) {
     }.toMap
   }
 
+  private def saveHadoopConf(): Unit = {
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:112]: Can org.apache.hadoop.conf.Configuration#writeXml(java.io.OutputStream) be used here？

Comment (turboFei) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:103]: thanks

File: kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala [hunk 2]
```diff
@@ -97,11 +100,14 @@ class MiniYarnService(name: String) extends AbstractService(name) {
...
+    val configToWrite = new Configuration(false)
     val hostName = InetAddress.getLocalHost.getHostName
-    yarnCluster.getConfig.iterator().asScala.map { kv =>
-      kv.getKey -> kv.getValue.replaceAll(hostName, "localhost")
-    }.toMap
+    yarnCluster.getConfig.iterator().asScala.foreach { kv =>
+      configToWrite.set(kv.getKey, kv.getValue.replaceAll(hostName, "localhost"))
+    }
+    val writer = new FileWriter(new File(hadoopConfDir, "yarn-site.xml"))
+    configToWrite.writeXml(writer)
```
Comment (pan3793) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:110]: will the `writeXml` close writer?

File: kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala [hunk 3]
```diff
@@ -97,11 +100,14 @@ class MiniYarnService(name: String) extends AbstractService(name) {
     super.stop()
```
Comment (ulysses-you) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:100]: Shall we delete the temporary file after stop ?

Comment (turboFei) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:101]: thanks, added.

The file created by Utils.createTempDir should be deleted on exit.

File: kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala (hunk 2 above)
Comment (turboFei) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/server/MiniYarnService.scala:111]: thanks, fixed.

**Review by pan3793 — APPROVED** [review]
LGTM, thanks!

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master for v1.3.0

**Pull request closed without merging**